pyvista/__init__.py

```diff
@@ -77,18 +77,20 @@
         raise FileNotFoundError(f'Invalid PYVISTA_USERDATA_PATH at {USER_DATA_PATH}')
 
 else:
-    # Set up data directory
-    USER_DATA_PATH = appdirs.user_data_dir('pyvista')
-    if not os.path.exists(USER_DATA_PATH):
-        os.makedirs(USER_DATA_PATH)
+    try:
+        # Set up data directory
+        USER_DATA_PATH = appdirs.user_data_dir('pyvista')
+        os.makedirs(USER_DATA_PATH, exist_ok=True)
+    except Exception as e:
+        warnings.warn(f'Unable to create `PYVISTA_USERDATA_PATH` at "{USER_DATA_PATH}"\n'
+                      f'Error: {e}\n\n'
+                      'Override the default path by setting the environmental variable '
+                      '`PYVISTA_USERDATA_PATH` to a writable path.')
+        USER_DATA_PATH = ''
```

Comment:
So if we cannot make a folder in `appdirs.user_data_dir('pyvista')`, this will attempt to make an examples folder in the current working directory? I'm not sure that is desirable, but maybe it is.

If not:  If the next try block raises, I think that `EXAMPLES_PATH` is unusable.  Does it make sense to harmonize here and just make `EXAMPLES_PATH` unusable if either try block fails?

Reply:
@MatthewFlamm I'm responsible for this suggestion :) I indeed just figured that harmonizing the two `makedirs`es would be better. But honestly, I don't really like the original behaviour where an unexpected error is just logged and ignored.

I don't know why it was written this way, so I didn't consider changing the old structure. But yeah, I would think that "unexpected error during `makedirs`" should log and then reraise.

Reply:
My thoughts are that we should have a single default location for this folder.  If that is not possible, it should be up to the user to specify it.  Having two possible locations is not ideal unless there are reasons I haven't thought of.

I don't think it should raise an error if this folder cannot be made as a lot of uses of this package will never use this folder.  It will be annoying to have to set an environment variable for functionality that is never used.  So I like this part of the implementation.

Reply:
Good point. Any chance that we could (or first "should") refactor so that these checks (and directory creation) are only done when those folders are used?

Reply:
I like that idea.  But that sounds like a different PR to me.

Reply:
Oh yes, absolutely. But if that's a viable option then we can just forget about my suggested changes here, and leave it for the refactor.

Reply:
I honestly didn't read the existing state too closely, so I'm okay with leaving all of this to a future PR.

Reply:
Sounds like this particular comment is resolved? I did move the two `os.path.join` outside the `try`

Reply:
I'll resolve this